numpy/lib/function_base.py

```diff
@@ -1912,10 +1911,16 @@ def diff(a, n=1, axis=-1):
     slice2[axis] = slice(None, -1)
     slice1 = tuple(slice1)
     slice2 = tuple(slice2)
+
+    if a.dtype == np.bool_:
+        da = a[slice1] ^ a[slice2]
```

Comment:
I note that at the ufunc level, `logical_xor` is defined as `not_equal` for booleans, but the operator `!=` goes through richcompare while `^` does not, hence `^` is a trifle faster.

One problem with the deprecation of `-` is that if we remove the boolean implementation the bools will be upcast to int8, which will really confuse people, but at least be consistent with python ...